File: packages/zuno_notifications/android/src/main/kotlin/im/zuno/chat/zuno_notifications/NotificationIds.kt
package im.zuno.chat.zuno_notifications

object NotificationIds {
    fun messageNotificationIdFor(roomId: String): Int {
        var hash = 0x811c9dc5.toInt()
        for (byte in roomId.toByteArray(Charsets.UTF_8)) {
            hash = hash xor (byte.toInt() and 0xff)
            hash *= 0x01000193
        }
        return hash and 0x7fffffff
    }
}

File: lib/core/notifications/notification_ids.dart
import 'dart:convert';

int fnv1a32(String input) {
  var hash = 0x811c9dc5;
  for (final byte in utf8.encode(input)) {
    hash ^= byte;
    hash = (hash * 0x01000193) & 0xffffffff;
  }
  return hash;
}

int messageNotificationIdFor(String roomId) => fnv1a32(roomId) & 0x7fffffff;
